### PR TITLE
added nano s/x signer (for real device and simulator)

### PR DESCRIPTION
**Pull request by shufps**
# Description of change

Added signers for Ledger Nano S/X hardware wallets (and simulator)

Signers can be enabled via features 'ledger-nano' and 'ledger-nano-simulator' and two signers are provided ('LedgerNanoSigner' and 'LedgerNanoSimulatorSigner').

### CODE REVIEW

**Comment by shufps** [comment]
Hmm, automatic building tests still fail - I'll have a look at it tomorrow :thinking:

**Comment by shufps** [comment]
ah I'm so stupid - forgot to readd the `ledger.rs` :see_no_evil: 

I try to update the PR

**Comment by shufps** [comment]
think it's ready to merge^^ :see_no_evil:

**Review by rajivshah3 — APPROVED** [review]

File: src/signing/mod.rs [hunk 1]
```diff
@@ -29,6 +32,12 @@ pub enum SignerType {
     #[cfg(feature = "stronghold")]
     #[cfg_attr(docsrs, doc(cfg(feature = "stronghold")))]
     Stronghold,
+    /// Ledger Device
+    #[cfg(feature = "ledger-nano")]
+    LedgerNanoSigner,
```
Comment (lucasfernog) [inline src/signing/mod.rs]: we should probably rename these to just `LedgerNano` and `LedgerNanoSimulator`, it feels redundant to have the "Signer" suffix

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
Nice work man!!!

File: src/signing/mod.rs (hunk 1 above)
Comment (shufps) [reply src/signing/mod.rs]: ok :) PR is still open, so I'll change it - and I also have to push a fix for the bip32 inputs that I missed to sort :see_no_evil:

**Comment by shufps** [comment]
> Nice work man!!!

I have to thank you! Without your great work, integrating into wallet.rs would have cost me weeks :see_no_evil:

**Pull request merged**